cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -728,6 +728,17 @@ func fixNodeGroupSize(context *context.AutoscalingContext, clusterStateRegistry
 					incorrectSize.ExpectedSize,
 					incorrectSize.CurrentSize,
 					delta)
+
+				possibleDelta := nodeGroup.MinSize() - incorrectSize.ExpectedSize
```

Comment:
Do we really want this check? MinSize is there to protect Cluster Autoscaler against scaling down below MinSize. If we want to fix the target size to sth below minSize, it means that the nodegroup started below minSize, we tried to scale up, but nodes are not registering correctly. I would expect that adjusting the target size of the group even to a value below minSize is something that we want to be able to do, we're not violating the minSize constraint, just reacting to a failed scale up situation.

Reply:
From the user's perspective, minsize is the minimum value of nodegroup target, and it cannot be less than minsize at any time.

On the other hand, many cloud vendors do not allow you to set a value lower than minsize.

Reply:
I don't think this makes sense in CA core. For cloud providers that prevent target size setting below min, adjusting the size can happen on cloud provider side. Otherwise it is going to prevent cloud providers that do allow going below min from working correctly. Note that incorrect size in CSR may stay incorrect if target size is adjusted up to match min size, so the subsequent loops will run this code again.